File: KRILI-frontend/src/app/models/etudiant-public-profil.dto.ts
// FICHIER: KRILI-frontend\src\app\models\etudiant-public-profil.dto.ts (ou un emplacement similaire)
export interface EtudiantPublicProfilDTO {
  id: number;
  nom: string;
  avatar?: string | null;
  etablissement?: string | null;
  filiere?: string | null;
  anneeEtude?: number | null;
  villeEtude?: string | null;
  avgRatingFromProprietaires?: number | null;
  reviewCountFromProprietaires?: number | null;
  // dateInscription?: string;
}

File: KRILI-backend/src/main/java/com/saadbarhrouj/krili/dto/EtudiantPublicProfilDTO.java
package com.saadbarhrouj.krili.dto;

import lombok.Data;
import java.math.BigDecimal; // Pour les stats d'avis

@Data
public class EtudiantPublicProfilDTO {
    private Long id;
    private String nom;
    private String avatar;
    private String etablissement;
    private String filiere;
    private Integer anneeEtude;
    private String villeEtude;

    private BigDecimal avgRatingFromProprietaires;
    private Integer reviewCountFromProprietaires;

}